inception-pdf-editor/src/main/java/de/tudarmstadt/ukp/inception/pdfeditor/pdfanno/model/PdfExtractFile.java

```diff
@@ -42,22 +45,94 @@
     private String stringContent;
 
     /**
-     * Contains position mapping for a character between PDFExtract string
+     * Contains position mapping for characters between PDFExtract string
```

Comment:
Is there an entry for *every character* in this map?

Reply:
Yes, the idea is to have a mapping of indices from the actual pdfextract file content with draw operations to indices from pdfextract file content without draw operations.
Then there is another map that maps indices from sanitized pdfextract file content without draw operations to the original pdfextract file content without draw operations.
This can be computed when a document is opened and while annotation we can just make lookups for the indices.

Reply:
I'd suggest using http://fastutil.di.unimi.it/docs/it/unimi/dsi/fastutil/ints/Int2IntMap.html here - it is not bidirectional, but since fastutil doesn't work with Integer objects but with primitives internally, it is probably still more efficient to use two of the fastutil maps than one BidiMap with Integer objects.